File: zad2/PamoLab/app/src/main/java/com/example/pamolab/ui/recipe/RecipeViewModel.java
package com.example.pamolab.ui.recipe;

import androidx.lifecycle.ViewModel;

import com.example.pamolab.model.Meal;

public class RecipeViewModel extends ViewModel {
    private Meal meal;

    public void setMeal(Meal meal) {
        this.meal = meal;
    }

    public String getMealName() {
        return meal.getName();
    }

    public String getIngredientsString() {
        StringBuilder sb = new StringBuilder();
        for (String ingredient : meal.getIngredients()) {
            sb.append(ingredient).append("\n");
        }
        return sb.toString();
    }

    public String getStepsString() {
        StringBuilder sb = new StringBuilder();
        int i = 1;
        for (String step : meal.getSteps()) {
            sb.append(i++).append(". ").append(step).append("\n");
        }
        return sb.toString();
    }
}

File: zad4/PamoLab/app/src/main/java/com/example/pamolab/ui/recipe/RecipeViewModel.kt
package com.example.pamolab.ui.recipe

import androidx.lifecycle.ViewModel
import com.example.pamolab.model.Meal

class RecipeViewModel : ViewModel() {
    private var meal: Meal? = null
    fun setMeal(meal: Meal?) {
        this.meal = meal
    }

    val mealName: String?
        get() = meal.getName()
    val ingredientsString: String
        get() {
            val sb = StringBuilder()
            for (ingredient in meal.getIngredients()) {
                sb.append(ingredient).append("\n")
            }
            return sb.toString()
        }
    val stepsString: String
        get() {
            val sb = StringBuilder()
            var i = 1
            for (step in meal.getSteps()) {
                sb.append(i++).append(". ").append(step).append("\n")
            }
            return sb.toString()
        }
}